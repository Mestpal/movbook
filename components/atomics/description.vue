<template>
  <v-card>
    <v-card-text>
      <ul
        class="contentDescription"
      >
        <li v-if="contentData.title">
          <h1 v-text="contentData.title" />
        </li>
        <li v-if="contentData.homepage">
          <a
            :href="contentData.homepage"
            target="_blank"
          >
            Homepage
          </a>
        </li>
        <li
          v-if="contentData.original_title"
          v-text="`Original title: ${contentData.original_title}`"
        />
        <li
          v-if="contentData.release_date"
          v-text="`Release date: ${contentData.release_date}`"
        />
        <li
          v-if="contentData.original_language"
          v-text="`Original language: ${contentData.original_language}`"
        />
        <li> Genres: </li>
        <ul
          class="contentListSecondLevel"
        >
          <li
            v-for="(genre, index) in contentData.genres"
            :key="index"
            v-text="genre.name"
          />
        </ul>
        <li> Production companies: </li>
        <ul
          v-if="contentData.production_companies && contentData.production_companies.length"
          class="contentListSecondLevel"
        >
          <li
            v-for="(company, index) in contentData.production_companies"
            :key="index"
            v-text="company.name"
          />
        </ul>
        <li
          v-if="contentData.overview"
        >
          <h2>Overview</h2>
          <p v-text="contentData.overview" />
        </li>
        <li>
          <h2>Vote average</h2>
          <p v-text="contentData.vote_average" />
        </li>
      </ul>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  props: {
    contentData: {
      type: Object,
      required: true,
      default: () => {}
    }
  }
}
</script>
