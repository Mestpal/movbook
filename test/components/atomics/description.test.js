
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'

import description from '@/components/atomics/description'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('description.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(description, {
      localVue,
      propsData: {
        contentData: {
          adult: false,
          backdrop_path: '/l5n4h4gmRtjfc0BMAXoJa6P9VC5.jpg',
          belongs_to_collection: null,
          budget: 0,
          genres: [
            { id: 28, name: 'Action' },
            { id: 35, name: 'Comedy' }
          ],
          homepage: '',
          id: 19133,
          imdb_id: 'tt0074205',
          original_language: 'en',
          original_title: 'The Big Bus',
          overview: "The ultimate disaster film parody. A nuclear powered bus is making it's maiden non-stop trip from New York to Denver. The journey is plagued by disasters due to the machinations of a mysterious group allied with the oil lobby. Will the down on his luck driver, with a reputation for eating his passengers, be able to complete the journey?",
          popularity: 5.348,
          poster_path: '/1DEKj25WaQvKQR1duZdXBJGVAFJ.jpg',
          production_companies: [
            {
              id: 4,
              logo_path: '/fycMZt242LVjagMByZOLUGbCvv3.png',
              name: 'Paramount',
              origin_country: 'US'
            }
          ],
          production_countries: [
            { iso_3166_1: 'US', name: 'United States of America' }
          ],
          release_date: '1976-06-23',
          revenue: 0,
          runtime: 88,
          spoken_languages: [
            { iso_639_1: 'en', name: 'English' }
          ],
          status: 'Released',
          tagline: 'The ride of your life.',
          title: 'The Big Bus',
          video: false,
          vote_average: 5.3,
          vote_count: 46
        }
      }
    })
  })

  it('reders the description component', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
