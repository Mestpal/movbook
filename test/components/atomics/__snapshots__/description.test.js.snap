// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description.vue reders the description component 1`] = `
"<div class=\\"v-card v-sheet theme--light\\">
  <div class=\\"v-card__text\\">
    <ul class=\\"contentDescription\\">
      <li>
        <h1>The Big Bus</h1>
      </li>
      <!---->
      <li>Original title: The Big Bus</li>
      <li>Release date: 1976-06-23</li>
      <li>Original language: en</li>
      <li> Genres: </li>
      <ul class=\\"contentListSecondLevel\\">
        <li>Action</li>
        <li>Comedy</li>
      </ul>
      <li> Production companies: </li>
      <ul class=\\"contentListSecondLevel\\">
        <li>Paramount</li>
      </ul>
      <li>
        <h2>Overview</h2>
        <p>The ultimate disaster film parody. A nuclear powered bus is making it's maiden non-stop trip from New York to Denver. The journey is plagued by disasters due to the machinations of a mysterious group allied with the oil lobby. Will the down on his luck driver, with a reputation for eating his passengers, be able to complete the journey?</p>
      </li>
      <li>
        <h2>Vote average</h2>
        <p>5.3</p>
      </li>
    </ul>
  </div>
</div>"
`;
